msg/templates/urtps/Publisher.cpp.em

```diff
@@ -111,12 +113,19 @@ bool @(topic)_Publisher::init()
 @[if ros2_distro]@
 @[    if ros2_distro == "ardent"]@
     Wparam.qos.m_partition.push_back("rt");
-    Wparam.topic.topicName = "@(topic)_PubSubTopic";
+    std::string topicName = ns;
+    topicName.append("@(topic)_PubSubTopic");
+    Wparam.topic.topicName = topicName;
 @[    else]@
-    Wparam.topic.topicName = "rt/@(topic)_PubSubTopic";
+    std::string topicName = "rt/";
+    topicName.append(ns);
+    topicName.append("@(topic)_PubSubTopic");
```

Comment:
the / is already appended to the ns already in parse_options:
 case 'n': if (nullptr != optarg) _options.ns = std::string(optarg) + "/"; break;

It is done like that so that if you don't give the namespace, you won't get double //

Reply:
ah yes sorry. Didn't saw that. It's a none-issue. Thanks